Add Google Weather docs (#39589)
home-assistant/home-assistant.github.io: source/_integrations/google_weather.markdown

---
title: Google Weather
description: Instructions on how to integrate Google Weather within Home Assistant.
ha_category:
  - Weather
ha_release: '2025.7'

> ha_iot_class: Cloud Polling

Author: What I have is correct.

ha_config_flow: true
ha_domain: google_weather
ha_codeowners:
  - '@tronikos'
ha_integration_type: service
ha_quality_scale: bronze
ha_platforms:
  - weather
---

This {% term integration %} uses [Google Weather](https://developers.google.com/maps/documentation/weather) as a source for weather data for your location.

## Prerequisites

You need an API key. Follow instructions [here](https://developers.google.com/maps/documentation/weather/get-api-key) for creating an API key.

> The Weather API is in the pre-GA Preview stage, and there is no charge for API usage during Preview.
> For pricing once it's in GA see [here](https://developers.google.com/maps/documentation/weather/usage-and-billing).
> It will be free as long as you stay under 10,000 requests per month.

Reviewer: You can put this in an informational note. :)

```suggestion
{% note %}
The Weather API is in the pre-GA Preview stage, and there is no charge for API usage during Preview.
For pricing once it's in GA see [here](https://developers.google.com/maps/documentation/weather/usage-and-billing).
It will be free as long as you stay under 10,000 requests per month.
{% endnote %}
```

---
title: Google Weather
description: Instructions on how to integrate Google Weather within Home Assistant.
ha_category:
  - Weather

> ha_release: '2025.7'

Reviewer 2: ```suggestion
ha_release: '2025.12'
```
tbd when the integration is merged

ha_quality_scale: bronze
ha_platforms:
  - weather
---

This {% term integration %} uses [Google Weather](https://developers.google.com/maps/documentation/weather) as a source for weather data for your location.

## Prerequisites

You need an API key. Follow instructions [in the Google developer documentation](https://developers.google.com/maps/documentation/weather/get-api-key) for creating an API key.

{% note %}
The Weather API is in the pre-GA Preview stage, and there is no charge for API usage during Preview.
For pricing once it's in GA refer to the [billing section in the Google developer documentation](https://developers.google.com/maps/documentation/weather/usage-and-billing).

It will be free as long as you stay under 10,000 requests per month.
{% endnote %}

{% include integrations/config_flow.md %}

## Data updates

The integration fetches:

- current weather conditions every 15 minutes
- daily weather forecast for the next 10 days every 1 hour
- hourly weather forecast for the next 24 hours every 1 hour

This results in 4,464 requests per month, meaning you could have up to 2 locations and still stay under the 10,000 free usage cap.

## Known limitations

> - Weather forecast information isn't currently available in South Korea and Japan. See [here](https://support.google.com/websearch/answer/13687874).

Reviewer 2: ```suggestion
- Weather forecast information isn't currently available in South Korea and Japan. Refer to the [Google Help Center](https://support.google.com/websearch/answer/13687874).

```

## Troubleshooting

> - Enable debug logging and attach logs before opening an issue.

Reviewer 2: ```suggestion
- [Enable debug logging](https://www.home-assistant.io/docs/configuration/troubleshooting/#debug-logs-and-diagnostics) and attach logs before opening an issue.

```
